addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -42,8 +42,10 @@ const TableOfContent = publicWidget.Widget.extend({
         this.$target.css('top', isHorizontalNavbar ? position : '');
         this.$target.find('.s_table_of_content_navbar').css('top', isHorizontalNavbar ? '' : position + 20);
         const $mainNavBar = $('#oe_main_menu_navbar');
-        position += $mainNavBar.length ? $mainNavBar.outerHeight() : 0;
+        const mainNavBarHidden = document.body.classList.contains('o_fullscreen') || this.editableMode;
```

Comment:
I would have said "what about checking the navbar is actually needed?" but that's true that in 16.0, we never have this kind of problem anymore so let's not worry about that :+1: